R/utils.R

```diff
@@ -1,190 +1,228 @@
+    res <- strsplit(x, "/", fixed = TRUE)[[1]]
+    if (endsWith(x, "/")) {
+        res[length(res)] <- paste0(res[length(res)], "/")
     }
     res
 }
 
+
 ## The path is a vector of names without bucket
-get_combined_path <- function(path_vec, is_folder){
-    combined_path <- paste0(path_vec,collapse="/")
-    if(is_folder&&length(path_vec)!=0&&!endsWith(combined_path,"/"))
-        combined_path <- paste0(combined_path,"/")
+get_combined_path <- function(path_vec, is_folder) {
+    combined_path <- paste0(path_vec, collapse = "/")
+    if (is_folder &&
+        length(path_vec) != 0 &&
+        !endsWith(combined_path, "/")) {
+        combined_path <- paste0(combined_path, "/")
+    }
     combined_path
 }
 
+
 ## Convert size in byte to a character format for print
-printable_size <-function(size_list){
-    result <- rep("",length(size_list))
-    ind <- size_list<10^3
-    result[ind] <- paste0(size_list[ind],"B")
-    ind <- size_list>=10^3&size_list<10^6
-    result[ind] <- paste0(myCeiling(size_list[ind]/10^3,digit=1),"KB")
-    ind <- size_list>=10^6&size_list<10^9
-    result[ind] <- paste0(myCeiling(size_list[ind]/10^6,digit=1),"MB")
-    ind <- size_list>=10^9&size_list<10^12
-    result[ind] <- paste0(myCeiling(size_list[ind]/10^9,digit=1),"GB")
-    ind <- size_list>=10^12
-    result[ind] <- paste0(myCeiling(size_list[ind]/10^12,digit=1),"TB")
+printable_size <- function(size_list) {
+    result <- rep("", length(size_list))
+
+    ind <- size_list < 10^3
+    result[ind] <- paste0(size_list[ind], "B")
+
+    ind <- size_list >= 10^3 & size_list < 10^6
+    result[ind] <- paste0(my_ceiling(size_list[ind] / 10^3, digit = 1), "KB")
+
+    ind <- size_list >= 10^6 & size_list < 10^9
+    result[ind] <- paste0(my_ceiling(size_list[ind] / 10^6, digit = 1), "MB")
+
+    ind <- size_list >= 10^9 & size_list < 10^12
+    result[ind] <- paste0(my_ceiling(size_list[ind] / 10^9, digit = 1), "GB")
+
+    ind <- size_list >= 10^12
+    result[ind] <- paste0(my_ceiling(size_list[ind] / 10^12, digit = 1), "TB")
+
     result
 }
 
-is_google_uri <- function(x){
-    if(startsWith(x,"gcs://")){
+
+is_google_uri <- function(x) {
+    if (startsWith(x, "gcs://")) {
         stop("The URI should start with `gs://`")
     }
-    startsWith(x,"gs://")
+    startsWith(x, "gs://")
 }
+
+
 get_google_URI <- function(bucket, file, full_path_vector = NULL) {
-    if(!is.null(full_path_vector)){
+    if (!is.null(full_path_vector)) {
         bucket <- full_path_vector[1]
         file <- full_path_vector[-1]
     }
-    paste0("gs://", bucket, "/", paste0(file,collapse="/"))
+    paste0("gs://", bucket, "/", paste0(file, collapse = "/"))
 }
-decompose_google_URI<-function(x, is_folder = NULL){
-    if(is_google_uri(x)){
-        x <- substring(x,first = 6)
-    }else{
+
+
+decompose_google_URI <- function(x, is_folder = NULL) {
+    if (is_google_uri(x)) {
+        x <- substring(x, first = 6)
+    } else {
+        ## FIX ME
```

Comment:
Not sure why there is an empty `else{}` clause here but this needs to be resolved as well.